src/istore.c

```diff
@@ -976,3 +976,73 @@ istore_svals(PG_FUNCTION_ARGS)
 
     SRF_RETURN_DONE(funcctx);
 }
+
+/*
+ * A boolean function that supports `select '1=>1'::istore = '2=>2'::istore`
+ */
+PG_FUNCTION_INFO_V1(istore_eq);
+Datum
+istore_eq(PG_FUNCTION_ARGS)
+{
+    IStore *is1, *is2;
+    IStorePair *pairs1, *pairs2;
+    int32 len;
+
+    if (PG_ARGISNULL(0) || PG_ARGISNULL(1)) PG_RETURN_NULL();
+
+    is1 = PG_GETARG_IS(0);
+    is2 = PG_GETARG_IS(1);
+
+    if (is1->len != is2->len) PG_RETURN_BOOL(false);
+    len = is1->len;
+
+    pairs1 = FIRST_PAIR(is1, IStorePair);
+    pairs2 = FIRST_PAIR(is2, IStorePair);
+
+    PG_RETURN_BOOL(memcmp(pairs1, pairs2, sizeof(IStorePair) * len) == 0);
+}
+
+/*
+ * Negation to istore_eq
+ */
+PG_FUNCTION_INFO_V1(istore_not_eq);
+Datum
+istore_not_eq(PG_FUNCTION_ARGS)
+{
+    PG_RETURN_BOOL(!DirectFunctionCall2(istore_eq, PG_GETARG_DATUM(0), PG_GETARG_DATUM(1)));
```

Comment:
@rapimo in this case this works because the result from `DirectFunctionCall2` is either 1 or 0, so the `DatumGetBool` is not required although it does look clearer. 

However, I get it you prefer an independent implementation here instead of just negating an `istore_eq` call?